platform_api/cluster_config_factory.py

```diff
@@ -134,20 +134,14 @@ def _create_tpu_preset(
         )
 
     def _create_resource_pool_type(self, payload: dict[str, Any]) -> ResourcePoolType:
-        cpu = payload.get("cpu")
-
-        memory = _get_memory_with_deprecated_mb(payload, "memory")
-        available_memory = _get_memory_with_deprecated_mb(payload, "available_memory")
         return ResourcePoolType(
             name=payload["name"],
             nvidia_gpu=payload.get("nvidia_gpu"),
             amd_gpu=payload.get("amd_gpu"),
             is_preemptible=payload.get("is_preemptible"),
-            cpu=cpu,
-            available_cpu=payload.get("available_cpu") or cpu,
-            memory=memory,
-            available_memory=available_memory or memory,
-            disk_gb=payload.get("disk_size_gb"),
+            cpu=payload.get("available_cpu") or payload.get("cpu"),
+            memory=payload.get("available_memory") or payload.get("memory"),
+            disk_size=payload.get("disk_size"),
```

Comment:
nit, for backward compatibility,

```suggestion
            disk_size=payload.get("disk_size") or payload.get("disk_size_gb")
```

Reply:
Actually this is not needed since disk_size was added in platform-config long time ago, it should be fine to just use disk_size